ksql-serde/src/main/java/io/confluent/ksql/serde/avro/KsqlAvroSerdeFactory.java

```diff
@@ -32,80 +32,75 @@
 import java.util.Objects;
 import java.util.function.Supplier;
 import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
 import org.apache.kafka.common.serialization.Serializer;
 import org.apache.kafka.connect.data.ConnectSchema;
 
 @Immutable
-public class KsqlAvroSerdeFactory extends KsqlSerdeFactory {
+public class KsqlAvroSerdeFactory implements KsqlSerdeFactory {
 
   private final String fullSchemaName;
 
   public KsqlAvroSerdeFactory(final String fullSchemaName) {
-    super(Format.AVRO);
     this.fullSchemaName = Objects.requireNonNull(fullSchemaName, "fullSchemaName").trim();
     if (this.fullSchemaName.isEmpty()) {
       throw new IllegalArgumentException("the schema name cannot be empty");
     }
   }
 
   @Override
-  public void validate(final ConnectSchema schema) {
-    // Supports all types
+  public Format getFormat() {
+    return Format.AVRO;
   }
 
   @Override
-  public boolean equals(final Object o) {
-    if (this == o) {
-      return true;
-    }
-    if (o == null || getClass() != o.getClass()) {
-      return false;
-    }
-    if (!super.equals(o)) {
-      return false;
-    }
-    final KsqlAvroSerdeFactory that = (KsqlAvroSerdeFactory) o;
-    return Objects.equals(fullSchemaName, that.fullSchemaName);
-  }
-
-  @Override
-  public int hashCode() {
-    return Objects.hash(super.hashCode(), fullSchemaName);
+  public void validate(final ConnectSchema schema) {
+    // Supports all types
   }
 
   @Override
-  protected Serializer<Object> createSerializer(
+  public Serde<Object> createSerde(
       final PersistenceSchema schema,
       final KsqlConfig ksqlConfig,
       final Supplier<SchemaRegistryClient> schemaRegistryClientFactory
   ) {
-    final Supplier<Serializer<Object>> supplier = () -> createConnectSerializer(
+    final Supplier<Serializer<Object>> serializerSupplier = () -> createConnectSerializer(
         schema,
         ksqlConfig,
         schemaRegistryClientFactory
     );
 
-    // Sanity check:
-    supplier.get();
-
-    return new ThreadLocalSerializer<>(supplier);
-  }
-
-  @Override
-  protected Deserializer<Object> createDeserializer(
-      final PersistenceSchema schema,
-      final KsqlConfig ksqlConfig,
-      final Supplier<SchemaRegistryClient> schemaRegistryClientFactory
-  ) {
-    final Supplier<Deserializer<Object>> supplier = () -> createConnectDeserializer(
+    final Supplier<Deserializer<Object>> deserializerSupplier = () -> createConnectDeserializer(
         schema,
         ksqlConfig,
         schemaRegistryClientFactory);
 
     // Sanity check:
-    supplier.get();
+    serializerSupplier.get();
+    deserializerSupplier.get();
+
+    return Serdes.serdeFrom(
+        new ThreadLocalSerializer<>(serializerSupplier),
+        new ThreadLocalDeserializer<>(deserializerSupplier)
+    );
+  }
+
+  @Override
+  public boolean equals(final Object o) {
```

Comment:
just wondering - do these factories actually need equals and hashcode implemented?

Reply:
only this one for some of the tests.